File: app/graphql/mutations/update_post_mutation.rb
Mutations::UpdatePostMutation = GraphQL::Relay::Mutation.define do
  name  'UpdatePost'

  input_field :id, !types.ID 
  input_field :title, types.String
  input_field :description, types.String
  input_field :image, Types::ImageType

  return_field :post, Types::PostType
  
  resolve ->(object, inputs, ctx) {
    user = ctx[:viewer].user

    attrs = {
      viewer: ctx[:viewer],
      creatorId: user.id,
      id: inputs[:id]
    }

    attrs[:title] = inputs[:title] if inputs.keys.include? "title"
    attrs[:description] = inputs[:description] if inputs.keys.include? "description"
    attrs[:image] = inputs[:image] if inputs.keys.include? "image"

    update_post = API::UpdatePost.call(attrs)

    if update_post.success?
      { post: update_post.post }
    else
      GraphQL::ExecutionError.new(update_post.error)
    end
  }
end

File: client/mutation/UpdatePostMutation.js
import { commitMutation, graphql } from 'react-relay'

const mutation = graphql`
  mutation UpdatePostMutation($input: UpdatePostInput!) {
    updatePost(input: $input) {
      post {
        id
        title
        description
        image
      }
    }
  }
`

function commit({ environment, input, onCompleted, onError }) {
  const variables = { input }

  commitMutation(environment, {
    mutation,
    variables,
    onCompleted,
    onError,
  })
}

export default {
  commit,
}
